pymongo/logger.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import enum
+import os
+from typing import Any
+
+from bson import UuidRepresentation, json_util
+from bson.json_util import JSONOptions
+
+
+class LogMessageStatus(str, enum.Enum):
+    STARTED = "Command started"
+    SUCCEEDED = "Command succeeded"
+    FAILED = "Command failed"
+
+
+DEFAULT_DOCUMENT_LENGTH = 1000
+_SENSITIVE_COMMANDS = [
+    "authenticate",
+    "saslStart",
+    "saslContinue",
+    "getnonce",
+    "createUser",
+    "updateUser",
+    "copydbgetnonce",
+    "copydbsaslstart",
+    "copydb",
+]
+_HELLO_COMMANDS = ["hello", "ismaster", "isMaster"]
+_REDACTED_FAILURE_FIELDS = ["code", "codeName", "errorLabels"]
+_DOCUMENTS = ["command", "reply", "failure"]
+_JSON_OPTIONS = JSONOptions(uuid_representation=UuidRepresentation.STANDARD)
+
+
+class LogMessage:
```

Comment:
Suggest adding `__slots__` for better perf.

Reply:
Is `__slots__` still useful if there are no public-facing methods? The intended use of `LogMessage` is to have it call `__str__` only.

Reply:
Yes it will reduce the memory footprint of instantiating the LogMessage class.

Reply:
Got it, so just make all the class methods slots?

Reply:
Not quite. slots is for the properties (self.kwargs in this case), not methods. Check out how we use slots on other classes like DatetimeMS: https://github.com/mongodb/mongo-python-driver/blob/6537415da70c2a15cad52ebe7669673947ace65a/bson/datetime_ms.py#L42